# Makes the MinGW build in Cygwin not interfering with the Cygwin itself.
diff --git config_office/set_soenv.in config_office/set_soenv.in
index 967e9f7..a4d2086 100644
--- config_office/set_soenv.in
+++ config_office/set_soenv.in
@@ -1175,8 +1175,7 @@ if ($platform =~ m/cygwin|os2/)
                         $wps.$JAVA_HOME.$LIB;
       if ( $USE_MINGW eq "cygwin" )
       {  $ILIB    .= $wps.PathFormat($USR.$LIB.$ds."mingw").
-                     $wps.PathFormat($USR.$LIB.$ds."w32api").
-                     $wps.PathFormat($USR.$LIB);
+                     $wps.PathFormat($USR.$LIB.$ds."w32api");
       }
       else
       {  $ILIB       .= $wps.$COMPATH.$LIB;
@@ -1553,6 +1552,7 @@ elsif ($platform =~ m/cygwin/)
    {  $SOLARINC       .= $I.'$SOLARVER'.$ds.'$INPATH'.$INC.$ds."external".$ds."mingw".$ds."include";
      if ( $USE_MINGW eq "cygwin" )
      {  $SOLARINC    .= $I.PathFormat($USR.$ds."include".$ds."mingw".$ds."include").
+                        $I.PathFormat($USR.$ds."include".$ds."mingw").
                         $I.PathFormat($USR.$ds."include".$ds."w32api").
                         $I.PathFormat($USR.$ds."include"); }
      else
